app/src/main/java/nsit/app/com/nsitapp/CalculatorActivity.java

```diff
@@ -42,263 +72,276 @@ public void onClick(View v) {
+                        VS2textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                    } else if (semSpinner.getSelectedItem().toString().equals("Sem 2")) {
 
 
-                        tc10.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                        tc11.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                        tc12.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                    } else if (s2.getSelectedItem().toString().equals("Sem 3")) {
+                        //PR5textView.setTextColor(getResources().getColor(R.color.calculator_notselected));
+                        PR5textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                        //VS1textView.setTextColor(getResources().getColor(R.color.calculator_notselected));
+                        VS1textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                        //VS2textView.setTextColor(getResources().getColor(R.color.calculator_notselected));
+                        VS2textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                    } else if (semSpinner.getSelectedItem().toString().equals("Sem 3")) {
 
 
-                        tc11.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                        tc12.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                    } else if (s2.getSelectedItem().toString().equals("Sem 4")) {
+                        VS1textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                        //VS2textView.setTextColor(getResources().getColor(R.color.calculator_notselected));
+                        VS2textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                    } else if (semSpinner.getSelectedItem().toString().equals("Sem 4")) {
 
-                    } else if (s2.getSelectedItem().toString().equals("Sem 5")) {
+                    } else if (semSpinner.getSelectedItem().toString().equals("Sem 5")) {
 
-                        tc10.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                        tc9.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                        tc12.setTextColor(getResources().getColor(R.color.calculator_notselected));
-                    } else if (s2.getSelectedItem().toString().equals("Sem 6")) {
+                        PR5textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
+                        PR4textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));
```

Comment:
This line `textView.setTextColor(ContextCompat.getColor(getContext(),R.color.calculator_notselected));` is being repeated a lot. Why don't you create a function that do this.

A function that takes 2 arguments : textview and color, and performs the task.